arrows/ocv/image_container.cxx

```diff
@@ -49,23 +49,28 @@ image_container
 ::image_container(const cv::Mat& d, ColorMode cm)
   : data_(d)
 {
-  if(cm != RGB)
+  //Handle BGR(A) case, others just copy the memory, the assumption is
+  //RGB images
+  if(cm == BGR && (data_.channels() == 3 || data_.channels() == 4 ))
   {
-    if ( data_.channels() == 3 )
+    switch(d.depth())
     {
-      cv::Mat new_color;
-      new_color.create( d.rows, d.cols, d.type() );
-
-      cv::cvtColor(data_, new_color, CV_BGR2RGB);
-      data_ = new_color;
-    }
-    else if ( data_.channels() == 4 )
-    {
-      cv::Mat new_color;
-      new_color.create( d.rows, d.cols, d.type() );
+      case CV_8U:
+      case CV_16U:
+      case CV_32F:
+      {
+        cv::Mat new_color;
+        new_color.create( d.rows, d.cols, d.type() );
 
-      cv::cvtColor(data_, new_color, CV_BGRA2RGBA);
-      data_ = new_color;
+        cv::cvtColor(data_, new_color, (data_.channels() == 3)?CV_BGR2RGB:CV_BGRA2RGBA);
+        data_ = new_color;
+        break;
+      }
+      case CV_8S:
```

Comment:
I think it would be better to use `default:` here in case you haven't covered all unsupported cases or more cases are added in the future.

Reply:
technically you don't need all those empty cases before the `default` because the default will cover them too, but this is a very minor point.  It's okay to merge without changing this.